File: 62_unique_paths.js
function uniquePaths(m, n) {
  if (m > 100 || n > 100) {
    console.log("Error: m and n must be at most 100.");
    return 0;
  }

  let count = 0; // To count all valid paths

  function backtrack(x, y) {
    // If we reached bottom-right corner
    if (x === m - 1 && y === n - 1) {
      count++;
      return;
    }

    // Move Right if possible
    if (x + 1 < m) {
      backtrack(x + 1, y);
    }

    // Move Down if possible
    if (y + 1 < n) {
      backtrack(x, y + 1);
    }
  }

  // Start from (0, 0) top-left corner
  backtrack(0, 0);
  return count;
}

// ----------- User Input (for Node.js) -----------

const readline = require("readline").createInterface({
  input: process.stdin,
  output: process.stdout,
});

readline.question("Enter number of columns (m): ", (mInput) => {
  readline.question("Enter number of rows (n): ", (nInput) => {
    const m = parseInt(mInput);
    const n = parseInt(nInput);

    console.log();

    if (m <= 100 && n <= 100) {
      console.log("Total Unique Paths:", uniquePaths(m, n));
    } else {
      console.log("Invalid input! Both m and n must be at most 100.");
    }

    readline.close();
  });
});

File: 62_unique_paths.py
def uniquePaths(m, n):
    if m > 100 or n > 100:
        print("Error: m and n must be at most 100.")
        return 0

    count = 0  # To count all valid paths

    def backtrack(x, y):
        nonlocal count
        # If we reached bottom-right corner
        if x == m - 1 and y == n - 1:
            count += 1
            return
        
        # Move Right if possible
        if x + 1 < m:
            backtrack(x + 1, y)
        
        # Move Down if possible
        if y + 1 < n:
            backtrack(x, y + 1)

    # Start from (0, 0) top-left corner
    backtrack(0, 0)
    return count

# ---------- User Input ----------
m = int(input("Enter number of columns (m): "))
n = int(input("Enter number of rows (n): "))

print()

if m <= 100 and n <= 100:
    print(f"Total Unique Paths: {uniquePaths(m, n)}")
else:
    print("Invalid input! Both m and n must be at most 100.")

